xdsl/pattern_rewriter.py

```diff
@@ -726,7 +731,11 @@ def _handle_operation_removal(self, op: Operation) -> None:
         """Handle removal of an operation."""
         if self.apply_recursively:
             self._add_operands_to_worklist(op.operands)
-        self._worklist.remove(op)
+        if op.regions:
```

Comment:
Is this a separate change to making `PatternRewriter` a `Builder`?

Reply:
I will rebase this branch on a new one, this was indeed a bug before that can be refactored.